dashboard/app/assets/javascripts/components/dialog.jsx

```diff
@@ -0,0 +1,59 @@
+window.dashboard = window.dashboard || {};
+
+/**
+ * Dialog
+ * A generic modal dialog that has an x-close in the upper right, and a
+ * semi-transparent backdrop. Can be closed by clicking the x, clicking the
+ * backdrop, or pressing esc.
+ */
+window.dashboard.Dialog = (function (React) {
+  return React.createClass({
+    propTypes: {
+      children: React.PropTypes.element.isRequired,
+      isOpen: React.PropTypes.bool.isRequired,
+      handleClose: React.PropTypes.func.isRequired
+    },
+
+    componentDidMount: function () {
+      this.focusDialog();
+    },
+
+    componentDidUpdate: function () {
+      this.focusDialog();
+    },
+
+    closeOnEscape: function (event) {
+      if (event.key === 'Escape') {
```

Comment:
React uses simulated events, which should have cross browser support https://facebook.github.io/react/docs/events.html